### ISSUE TITLE
Empty Time Series Groups list

### ISSUE THREAD

**Issue by snzhao-usace** [opendcs/opendcs#850]
**Describe the bug**
Time Series Groups list is empty.   Same issue happens at all 3 DBs, NWS/NWP/NWW.   

**To Reproduce**
1. Start OpenDCS 7.0
2. Click on "Time Series Groups"
3. Log in and window will open
4. List is empty

**Expected behavior**
See existing time serious groups

**Screenshots**
![Screenshot 2024-12-15 090848](https://github.com/user-attachments/assets/97683326-1663-4cf4-b8a7-3cb6278618b4)

**Desktop/ServerSide (please complete the following information):**
 - Windows 11
 - OpenDCS 7.0.13

**Additional context**
It seems a bug with OpenDCS Client.

**Comment by MikeNeilson** [comment]
Well, that's a strange one. the ultimate error:

```
java.lang.IllegalArgumentException: Comparison method violates its general contract!
        at java.base/java.util.TimSort.mergeHi(TimSort.java:903)
        at java.base/java.util.TimSort.mergeAt(TimSort.java:520)
        at java.base/java.util.TimSort.mergeCollapse(TimSort.java:448)
        at java.base/java.util.TimSort.sort(TimSort.java:245)
        at java.base/java.util.Arrays.sort(Arrays.java:1307)
        at java.base/java.util.ArrayList.sort(ArrayList.java:1721)
        at java.base/java.util.Collections.sort(Collections.java:179)
        at decodes.tsdb.groupedit.TsGroupsSelectTableModel.sortByColumn(TsGroupListPanel.java:541)
        at decodes.tsdb.groupedit.TsGroupsSelectTableModel.setTsGroupListFromDb(TsGroupListPanel.java:350)
        at decodes.tsdb.groupedit.TsGroupListPanel.setTsGroupListFromDb(TsGroupListPanel.java:272)
        at decodes.tsdb.groupedit.TsDbGrpListPanel.jbInit(TsDbGrpListPanel.java:125)
        at decodes.tsdb.groupedit.TsDbGrpListPanel.<init>(TsDbGrpListPanel.java:95)
        at decodes.tsdb.groupedit.TsDbGrpEditorFrame.jbInit(TsDbGrpEditorFrame.java:161)
        at decodes.tsdb.groupedit.TsDbGrpEditorFrame.<init>(TsDbGrpEditorFrame.java:84)
        at decodes.tsdb.groupedit.TsDbGrpEditor.runApp(TsDbGrpEditor.java:116)
        at decodes.tsdb.TsdbAppTemplate.execute(TsdbAppTemplate.java:180)
        at decodes.tsdb.groupedit.TsDbGrpEditor.main(TsDbGrpEditor.java:200)
```

I'm pretty sure we haven't changed any code that would result in that type of error but it's always possible you managed to hit an edge case with your data (which is not a statement that you've done anything wrong).

I wasn't able to recreate the error with the NWDM data, which is of near equal complexity.

I know you can't get in a look but please let us know, at least roughly, what has been recently changed or added. That will help with investigating what's going wrong.

**Comment by MikeNeilson** [comment]
Nevermind, about the what. It's just that you added something. The function that handles the sorting was assuming ID fields that fit within an Integer; however, the CWMS _code fields need a Long. Unfortunately you've managed to have that sequence finally count up enough to hit the limit.

Unfortunately there's nothing we can do immediately and this will likely affect all previous versions as well. 

Goodish news is it's limited to the GUI itself so compproc, et all, should continue to run unimpeded.
Other good news is it's a simple fix that's already done.

Bad news is, we're not even going to think about starting another release cycle until after the new year starts.

**Comment by MikeNeilson** [comment]
However, I will port this immediately to the 7.0 branch, so you can use the nightly build (https://github.com/opendcs/opendcs/releases/tag/7.0-nightly) in the mean time for this purpose. 

The nightly on the release branch *should* be stable and have no data destructive issues, but fair warning, the are not vetting in any way, just created, and we only started that process last week. Use with caution.

**Comment by snzhao-usace** [comment]
I tried and nightly build and still don’t see any time series group listed.


  1.  I re-named the existing “C:\OpenDCS” folder a different name, “C:\Zhao_OpenDCS”
  2.  Created a folder “C:\OpenDCS”
  3.  Copied downloaded “opendcs-7.99.7.0-SNAPSHOT.tar” to OpenDCS
  4.  Double clicked tar file, and go to folder “opendcs-7.99.7.0-SNAPSHOT”
  5.  Copied all files/folders in “opendcs-7.99.7.0-SNAPSHOT” to “OpenDCS”
  6.  Launched OpenDCS, I am able to log in to NWS T7, but still got an empty time series groups list

Not sure if any steps are missing.   Thanks.



From: Mike Neilson ***@***.***>
Sent: Monday, December 16, 2024 7:56 AM
To: opendcs/opendcs ***@***.***>
Cc: Zhao, Shilong N CIV USARMY CENWS (USA) ***@***.***>; Author ***@***.***>
Subject: [Non-DoD Source] Re: [opendcs/opendcs] Empty Time Series Groups list (Issue #850)


However, I will port this immediately to the 7.0 branch, so you can use the nightly build (https://github.com/opendcs/opendcs/releases/tag/7.0-nightly<BLOCKEDhttps://github.com/opendcs/opendcs/releases/tag/7.0-nightly>) in the mean time for this purpose.

The nightly on the release branch should be stable and have no data destructive issues, but fair warning, the are not vetting in any way, just created, and we only started that process last week. Use with caution.

**Comment by MikeNeilson** [comment]
The fix has not yet been merged into the 7.0 branch and thus would not be available in yesterdays nightly build.

Once the changes have been merged in I will trigger an update to the nightly; check for when the date on the file is more recent than now.

**Comment by snzhao-usace** [comment]
I tried the latest night build this morning and I am able to see all time series groups.   Thanks a lot for your help.


From: Mike Neilson ***@***.***>
Sent: Monday, December 16, 2024 10:09 AM
To: opendcs/opendcs ***@***.***>
Cc: Zhao, Shilong N CIV USARMY CENWS (USA) ***@***.***>; Author ***@***.***>
Subject: [Non-DoD Source] Re: [opendcs/opendcs] Empty Time Series Groups list (Issue #850)


The fix has not yet been merged into the 7.0 branch and thus would not be available in yesterdays nightly build.

Once the changes have been merged in I will trigger an update to the nightly; check for when the date on the file is more recent than now.
